f600233d1b154d507b782f84604bcb12/update/sys_hub_action_type_definition_7232d6c92fe41110e58359a72799b6f0.xml

```diff
@@ -0,0 +1,1331 @@
+        <sys_mod_count>4</sys_mod_count>
+        <sys_scope display_value="ATAT">f600233d1b154d507b782f84604bcb12</sys_scope>
+        <sys_updated_by>jeff.segal-ctr@ccpo.mil</sys_updated_by>
+        <sys_updated_on>2022-07-06 02:09:15</sys_updated_on>
+    </sys_hub_step_instance>
+    <sys_variable_value action="delete_multiple" query="document_key=4d88d6412f281110e58359a72799b6de"/>
+    <sys_variable_value action="INSERT_OR_UPDATE">
+        <document>sys_hub_step_instance</document>
+        <document_key>4d88d6412f281110e58359a72799b6de</document_key>
+        <order>400</order>
+        <sys_class_name>sys_variable_value</sys_class_name>
+        <sys_created_by>jeff.segal-ctr@ccpo.mil</sys_created_by>
+        <sys_created_on>2022-07-06 02:05:54</sys_created_on>
+        <sys_id>1d88d6412f281110e58359a72799b6fa</sys_id>
+        <sys_mod_count>0</sys_mod_count>
+        <sys_updated_by>jeff.segal-ctr@ccpo.mil</sys_updated_by>
+        <sys_updated_on>2022-07-06 02:05:54</sys_updated_on>
+        <value>35aa573fd7802200bdbaee5b5e610375</value>
+        <variable display_value="MID Application">f5e56d79b3101300176b051a16a8dce4</variable>
+    </sys_variable_value>
+    <sys_variable_value action="INSERT_OR_UPDATE">
+        <document>sys_hub_step_instance</document>
+        <document_key>4d88d6412f281110e58359a72799b6de</document_key>
+        <order>600</order>
+        <sys_class_name>sys_variable_value</sys_class_name>
+        <sys_created_by>jeff.segal-ctr@ccpo.mil</sys_created_by>
+        <sys_created_on>2022-07-06 02:05:54</sys_created_on>
+        <sys_id>dd88d6412f281110e58359a72799b6fa</sys_id>
+        <sys_mod_count>3</sys_mod_count>
+        <sys_updated_by>jeff.segal-ctr@ccpo.mil</sys_updated_by>
+        <sys_updated_on>2022-07-06 02:09:15</sys_updated_on>
+        <value>function addCostRecord(result, csp, organization, portfolio, taskOrder, clin, isActual) {
+  var gdt = new GlideDateTime();
+  gdt.setDisplayValue(result.month, "yyyy-MM");
+  
+  // Check to see if we already have a record for this Task Order / CLIN / Year-Month Tuple
+  var cost = new GlideRecord('x_g_dis_atat_costs');
+  cost.addQuery("task_order_number", taskOrder.task_order_number);
+  cost.addQuery("clin", clin.clin_number);
+  cost.addQuery("year_month", gdt.getDate());
+  cost.query();
+  
+  var isUpdate = false;
+  if (cost.hasNext()) {
+    cost.next();
+    isUpdate = true;       
+  }
+  else {
+    cost.newRecord();
+    isUpdate = false;       
+  }
+
+  cost.clin = clin.clin_number;
+  cost.csp = csp.sys_id;
+  cost.is_actual = isActual;
+  cost.organization = organization.sys_id;
+  cost.portfolio = portfolio.sys_id;
+  cost.service_agency = organization.service_agency;
+  cost.task_order_number = taskOrder.task_order_number;
+  cost.value = result.value;
+  cost.year_month = gdt;
+  
+  if (isUpdate) {
+    cost.update();
+  }
+  else {
+    cost.insert();
+  }
+  
+}
+
+function handleCostObject(costObject) {
```

Comment:
I'm not sure if these functions planned to be used in other places, but it might be beneficial to make the `addCostRecord` and `handleCostObject` as script include functions as is to use else where and they can also be called in background scripts for testing purposes.

Reply:
I hadn't thought of that, but I think I'd rather defer that to a time in which we'd need to use these functions elsewhere because I'm not sure if we will. Ad hoc testing is pretty easy via the Action as-is.